mod_regression/controllers.py

```diff
@@ -114,11 +114,29 @@ def test_result(regression_id):
     pass
 
 
-@mod_regression.route('/test/new')
+@mod_regression.route('/test/new', methods=['GET', 'POST'])
+@template_renderer()
+@check_access_rights([Role.admin])
 def test_add():
-    # Add a new regression test
-    pass
-
+    """
+    Function to add a regression test
+    """
+    form = AddTestForm(request.form)
+    form.sample_id.choices = [(sam.id,sam.sha) for sam in Sample.query.all()]
+    form.category_id.choices = [(cat.id,cat.name) for cat in Category.query.all()]
+    if form.expected_rc.data < 0:
+        form.expected_rc.errors.append("Expected Runtime Code can't be negative")
+    elif form.submit.data and form.validate_on_submit():
```

Comment:
no need to check form.submit.data, it will be checked in form.validate_on_submit()